Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp

```diff
@@ -22,6 +25,72 @@ JS::NonnullGCPtr<CSSStyleSheet> CSSStyleSheet::create(JS::Realm& realm, CSSRuleL
     return realm.heap().allocate<CSSStyleSheet>(realm, realm, rules, media, move(location));
 }
 
+// https://drafts.csswg.org/cssom/#dom-cssstylesheet-cssstylesheet
+WebIDL::ExceptionOr<JS::NonnullGCPtr<CSSStyleSheet>> CSSStyleSheet::construct_impl(JS::Realm& realm, Optional<CSSStyleSheetInit> const& options)
+{
+    // 1. Construct a new CSSStyleSheet object sheet.
+    auto sheet = create(realm, CSSRuleList::create_empty(realm), CSS::MediaList::create(realm, {}), {});
+
+    // 2. Set sheet’s location to the base URL of the associated Document for the current global object.
+    auto associated_document = sheet->global_object().document();
+    sheet->set_location(MUST(associated_document->base_url().to_string()));
+
+    // 3. Set sheet’s stylesheet base URL to the baseURL attribute value from options.
+    if (options.has_value() && options->base_url.has_value()) {
+        Optional<AK::URL> sheet_location_url;
+        if (sheet->location().has_value())
+            sheet_location_url = sheet->location().release_value();
+
+        // AD-HOC: This isn't explicitly mentioned in the specification, but multiple modern browsers do this.
+        AK::URL url = sheet->location().has_value() ? sheet_location_url->complete_url(options->base_url.value()) : options->base_url.value();
+        if (!url.is_valid())
+            return WebIDL::NotAllowedError::create(realm, "Constructed style sheets must have a valid base URL"_fly_string);
```

Comment:
Throwing an error if an invalid base URL is given allows us to pass [this WPT test](https://wpt.live/css/cssom/CSSStyleSheet-constructable-baseURL.tentative.html). I wasn't sure if this was worth doing or not, since [not all browsers](https://wpt.fyi/results/?label=master&label=experimental&aligned&q=%2Fcss%2Fcssom%2Fcssstylesheet-constructable-baseurl.tentative.html) do it and it isn't explicitly mentioned in the specification.

Reply:
Seems fine :)